monai/metrics/surface_distance.py

```diff
@@ -0,0 +1,179 @@
+
+    Args:
+        seg_1: first binary or labelfield image.
+        seg_2: second binary or labelfield image.
+        label_idx: for labelfield images, convert to binary with
+            `seg_1 = seg_1 == label_idx`.
+        crop: crop input images and only keep the foregrounds. In order to
+            maintain two inputs' shapes, here the bounding box is achieved
+            by ``(seg_1 | seg_2)`` which represents the union set of two
+            images. Defaults to ``True``.
+    """
+
+    # Get both labelfields as np arrays
+    if torch.is_tensor(seg_1):
+        seg_1 = seg_1.detach().cpu().numpy()
+    if torch.is_tensor(seg_2):
+        seg_2 = seg_2.detach().cpu().numpy()
+
+    # Check non-zero number of elements and same shape
+    if seg_1.size == 0 or seg_1.shape != seg_2.shape:
+        raise ValueError("Labelfields should have same shape (and non-zero number of elements)")
+
+    # If not binary images, convert them
+    if seg_1.dtype != bool:
+        seg_1 = seg_1 == label_idx
+    if seg_2.dtype != bool:
+        seg_2 = seg_2 == label_idx
+
+    # Check both have at least 1 voxel with desired index
+    if not (np.any(seg_1) and np.any(seg_2)):
+        raise ValueError(f"Labelfields should have at least 1 voxel containing the desired labelfield, {label_idx}")
+
+    if crop:
+        seg_1, seg_2 = np.expand_dims(seg_1, 0), np.expand_dims(seg_2, 0)
+        box_start, box_end = generate_spatial_bounding_box(seg_1 | seg_2)
+        cropper = SpatialCrop(roi_start=box_start, roi_end=box_end)
+        seg_1, seg_2 = np.squeeze(cropper(seg_1)), np.squeeze(cropper(seg_2))
+
+    # Do binary erosion and use XOR to get edges
+    edges_1 = binary_erosion(seg_1) ^ seg_1
+    edges_2 = binary_erosion(seg_2) ^ seg_2
+
+    return (edges_1, edges_2)
+
+
+def get_surface_distance(
+    seg_1: Union[np.ndarray, torch.Tensor],
+    seg_2: Union[np.ndarray, torch.Tensor],
+    label_idx: int,
+    crop: bool = True,
+) -> np.ndarray:
+    """
+    This function is used to compute the surface distances from `seg_1` to `seg_2`.
+
+    In order to improve the computing efficiency, before getting the edges,
+    the images can be cropped and only keep the foreground if not specifies
+    ``crop = False``.
+
+    Args:
+        seg_1: first binary or labelfield image.
+        seg_2: second binary or labelfield image.
+        label_idx: for labelfield images, convert to binary with
+            `seg_1 = seg_1 == label_idx`.
+        crop: crop input images and only keep the foregrounds. In order to
+            maintain two inputs' shapes, here the bounding box is achieved
+            by ``(seg_1 | seg_2)`` which represents the union set of two
+            images. Defaults to ``True``.
+    """
+    (edges_1, edges_2) = get_mask_edges(seg_1, seg_2, label_idx, crop)
+    dis_1 = distance_transform_edt(~edges_2)
+    surface_distance_1 = dis_1[edges_1]
+    return surface_distance_1
+
+
+def compute_percent_hausdorff_distance(
```

Comment:
this could be a `compute_percent_hausdorff_distance` function in `monai.metrics.hausdorff_distance`:
```py
def compute_percent_hausdorff_distance(seg_1, seg_2, metric, percentile, label_idx, directed):
    if metric in ("chessboard, taxicab"):
         use the distance transform_edt: get_surface_distance(seg_1, seg_2, metric)
         return np.percentile(distance, percentile)
    if metric.lower() == "euclidean" and (percentile is None or percentile==100):
         use monai.metrics.hausdorff_distance.compute_hausdorff_distance
         return the metric
    if metric.lower() == "euclidean" and (0< percentile <100):
         need an implementation here...
```

Reply:
Thanks! I've modified all the places and updated the code.